app/services/bulk_hosts_manager.rb

```diff
@@ -19,6 +19,13 @@ def build(reboot: false)
     end
   end
 
+  def reassign_hostgroups(hostgroup)
+    @hosts.each do |host|
+      host.hostgroup = hostgroup
+      host.save(:validate => false)
+    end
```

Comment:
This is the legacy method used for this: https://github.com/theforeman/foreman/blob/b9c2dd00bed537b73bbad5ba529130e8717dfabb/app/controllers/hosts_controller.rb#L430

Reply:
Updated old controller to use the Bulk host manager method.